### PR TITLE
[Doc clarification] Added unidirectional text for LayerNorm

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
### Description
Doc only, added clarification that scale and B can be uni-directionally expanded to X. Current def is not explicit on this point.

### Motivation and Context
Clarification of docs to be more explicit. 
Issue discussed here #5666

### CODE REVIEW

File: docs/Changelog-ml.md [hunk 1]
```diff
@@ -227,7 +227,7 @@ This version of the operator has been available since version 1 of the 'ai.onnx.
 ### <a name="ai.onnx.ml.FeatureVectorizer-1"></a>**ai.onnx.ml.FeatureVectorizer-1**</a>
 
   Concatenates input tensors into one continuous output.<br>
-      All input shapes are 2-D and are concatenated along the second dimension. 1-D tensors are treated as [1,C].
+      All input shapes are 2-D and are concatenated along the second dimention. 1-D tensors are treated as [1,C].
```
Comment (justinchuby) [inline docs/Changelog-ml.md:230]: typo?

**Comment by justinchuby** [comment]
Looks like there’s some unrelated changes?

File: onnx/defs/nn/defs.cc [hunk 2]
```diff
@@ -2516,7 +2516,9 @@ static const char* LayerNormalization_ver17_doc = R"DOC(
       Let `d[i]` indicate the i-th dimension of `X`.
       If `X`'s shape is `[d[0], ..., d[axis-1], d[axis], ..., d[rank-1]]`,
       the shape of `Mean` and `InvStdDev` is `[d[0], ..., d[axis-1], 1, ..., 1]`.
-      `Y` and `X` have the same shape.
+      `Y` and `X` have the same shape. This operator supports unidirectional broadcasting
+      (tensors `Scale` and `B` should be unidirectional broadcastable to tensor `X`); 
+      for more details please check [the doc](Broadcasting.md).
```
Comment (justinchuby) [inline onnx/defs/nn/defs.cc:2521]:
```suggestion
      `Y` and `X` have the same shape. This operator supports unidirectional broadcasting
      (tensors `Scale` and `B` should be unidirectional broadcastable to tensor `X`);
      for more details please check [the doc](Broadcasting.md).
```

**Comment by justinchuby** [comment]
I added auto gen doc to see if it would fix the generated file issues.

**Review by justinchuby — APPROVED** [review]
cc @gramalingam

**Review by gramalingam — APPROVED** [review]

**Comment by AlexandreEichenberger** [comment]
The documentation indicated to run `tools/update_doc.sh` but that resulted in failures (in addition, still rely on `python` instead of `python3`. I then try to do a build, but since we are using a slightly different protobuf for onnx-mlir, did not want to install anything. I finally saw a suggestion to do a `python3 onnx/defs/gen_doc.py` which seems to have generated the right doc.

Being very new to doc update, it might be nice to have an entry somewhere in the doc "I just want to clarify a documentation issue, here is how to proceed". If you let me know what is the approved approach to that, happy to update the doc.

**Comment by justinchuby** [comment]
tools/update_doc.sh should work. Could you post the error?

**Comment by AlexandreEichenberger** [comment]
First it's still using `python` which I don't have. So I just `ln -s` python3, which might be the cause of the issues. Below is the log
```
alexe@AChampignon onnx % tools/update_doc.sh
===> recompile onnx
running develop
/opt/homebrew/lib/python3.11/site-packages/setuptools/command/develop.py:40: EasyInstallDeprecationWarning: easy_install command is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` and ``easy_install``.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://github.com/pypa/setuptools/issues/917 for details.
        ********************************************************************************

!!
  easy_install.initialize_options(self)
/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py:66: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
running egg_info
writing onnx.egg-info/PKG-INFO
writing dependency_links to onnx.egg-info/dependency_links.txt
writing entry points to onnx.egg-info/entry_points.txt
writing requirements to onnx.egg-info/requires.txt
writing top-level names to onnx.egg-info/top_level.txt
reading manifest file 'onnx.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no files found matching '*.c' under directory 'onnx'
warning: no files found matching '*' under directory 'third_party'
adding license file 'LICENSE'
writing manifest file 'onnx.egg-info/SOURCES.txt'
running build_ext
running cmake_build
Extra cmake args: ['-DONNX_USE_PROTOBUF_SHARED_LIBS=ON']
Using cmake args: ['/opt/homebrew/bin/cmake', '-DPYTHON_INCLUDE_DIR=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11', '-DPYTHON_EXECUTABLE=/opt/homebrew/opt/python@3.11/bin/python3.11', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cpython-311-darwin.so', '-DCMAKE_BUILD_TYPE=Release', '-DONNX_ML=1', '-DONNX_USE_PROTOBUF_SHARED_LIBS=ON', '/Users/alexe/Data/onnx/onnx']
-- ONNX_PROTOC_EXECUTABLE: /usr/local/bin/protoc
-- Protobuf_VERSION: 3.20.3
Generated: /Users/alexe/Data/onnx/onnx/.setuptools-cmake-build/onnx/onnx-ml.proto
Generated: /Users/alexe/Data/onnx/onnx/.setuptools-cmake-build/onnx/onnx-operators-ml.proto
Generated: /Users/alexe/Data/onnx/onnx/.setuptools-cmake-build/onnx/onnx-data.proto
-- Could NOT find pybind11 (missing: pybind11_DIR)
CMake Error at CMakeLists.txt:561 (message):
  cannot find pybind


-- Configuring incomplete, errors occurred!
Traceback (most recent call last):
  File "/Users/alexe/Data/onnx/onnx/setup.py", line 327, in <module>
    setuptools.setup(
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
    dist.run_commands()
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 1233, in run_command
    super().run_command(command)
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/Users/alexe/Data/onnx/onnx/setup.py", line 259, in run
    return super().run()
           ^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/command/develop.py", line 34, in run
    self.install_for_development()
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/command/develop.py", line 109, in install_for_development
    self.run_command('build_ext')
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 1233, in run_command
    super().run_command(command)
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/Users/alexe/Data/onnx/onnx/setup.py", line 264, in run
    self.run_command("cmake_build")
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
    self.distribution.run_command(command)
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/dist.py", line 1233, in run_command
    super().run_command(command)
  File "/opt/homebrew/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
    cmd_obj.run()
  File "/Users/alexe/Data/onnx/onnx/setup.py", line 235, in run
    subprocess.check_call(cmake_args)
  File "/opt/homebrew/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/subprocess.py", line 413, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/opt/homebrew/bin/cmake', '-DPYTHON_INCLUDE_DIR=/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11', '-DPYTHON_EXECUTABLE=/opt/homebrew/opt/python@3.11/bin/python3.11', '-DBUILD_ONNX_PYTHON=ON', '-DCMAKE_EXPORT_COMPILE_COMMANDS=ON', '-DONNX_NAMESPACE=onnx', '-DPY_EXT_SUFFIX=.cpython-311-darwin.so', '-DCMAKE_BUILD_TYPE=Release', '-DONNX_ML=1', '-DONNX_USE_PROTOBUF_SHARED_LIBS=ON', '/Users/alexe/Data/onnx/onnx']' returned non-zero exit status 1.
```

**Comment by justinchuby** [comment]
The error is `cannot find pybind`. Perhaps pybind was not in the search path?

**Comment by gramalingam** [comment]
Also worth checking if the git submodules have been updated, since pybind is a submodule, IIRC.

**Pull request merged**